deepchecks/core/serialization/common.py

```diff
@@ -0,0 +1,369 @@
+) -> pd.DataFrame:
+    """Return the conditions table as DataFrame.
+
+    Parameters
+    ----------
+    check_results : Union['CheckResult', List['CheckResult']]
+        check results to show conditions of.
+    max_info_len : int
+        max length of the additional info.
+    include_icon : bool , default: True
+        if to show the html condition result icon or the enum
+    include_check_name : bool, default False
+        whether to include check name into dataframe or not
+    output_id : str
+        the unique id to append for the check names to create links (won't create links if None/empty).
+
+    Returns
+    -------
+    pd.Dataframe:
+        the condition table.
+    """
+    check_results = (
+        [check_results]
+        if isinstance(check_results, check_types.CheckResult)
+        else check_results
+    )
+
+    data = []
+
+    for check_result in check_results:
+        for cond_result in check_result.conditions_results:
+            priority = cond_result.priority
+            icon = cond_result.get_icon() if include_icon else cond_result.category.value
+            check_header = check_result.get_header()
+
+            if output_id:
+                link = f'<a href=#{check_result.get_check_id(output_id)}>{check_header}</a>'
+            else:
+                link = check_header
+                # if it has no display show on bottom for the category (lower priority)
+                priority += 0.1
+
+            data.append([
+                icon, link, cond_result.name, cond_result.details, priority
+            ])
+
+    df = pd.DataFrame(
+        data=data,
+        columns=['Status', 'Check', 'Condition', 'More Info', 'sort']
+    )
+
+    df.sort_values(by=['sort'], inplace=True)
+    df.drop('sort', axis=1, inplace=True)
+
+    if include_check_name is False:
+        df.drop('Check', axis=1, inplace=True)
+
+    df['More Info'] = df['More Info'].map(lambda x: get_ellipsis(x, max_info_len))
+
+    with warnings.catch_warnings():
+        warnings.simplefilter(action='ignore', category=FutureWarning)
+        return df.style.hide_index()
+
+
+def requirejs_script(connected: bool = True):
+    """Return requirejs script tag.
+
+    Parameters
+    ----------
+    connected : bool, default True
+
+    Returns
+    -------
+    str
+    """
+    if connected is True:
+        return textwrap.dedent("""
+            <script
+                src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js"
+                integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg=="
+                crossorigin="anonymous"
+                referrerpolicy="no-referrer">
+            </script>
+        """)
+    else:
+        path = os.path.join('core', 'resources', "requirejs.min.js")
+        js = pkgutil.get_data('deepchecks', path).decode('utf-8')
+        return f'<script>{js}</script>'
+
+
+def plotlyjs_script(connected: bool = True) -> str:
```

Comment:
why is it suddenly needed?

Reply:
plotly plots do not work without plotlyjs javascript library

Reply:
but we didn't do anything for it and it worked

Reply:
yes, because we have been using IPython.display and Plotly API for plots display that handles this under the hood
but in some cases, we need more precise control over the output and need to deal with this by ourselves

Reply:
The plotlyjs is according to the installed plotly version?

Reply:
yes

```python
...
from plotly.io._utils import plotly_cdn_url
from plotly.offline.offline import get_plotlyjs
...
plotly_cdn=plotly_cdn_url().rstrip(".js")
script=get_plotlyjs()
...
```